src/ldap_helper.c

```diff
@@ -747,6 +739,25 @@ destroy_ldap_instance(ldap_instance_t **ldap_instp)
 	*ldap_instp = NULL;
 }
 
+void watcher_shutdown(ldap_instance_t *ldap_inst)
```

Comment:
Maybe this could be named `ldap_syncrepl_watcher_shutdown` so it is clear from its name where it belongs...